WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailActivity.java

```diff
@@ -252,15 +252,21 @@ private void showErrorToastAndFinish() {
         finish();
     }
 
-    private void markNoteAsRead(Note note) {
-        GCMMessageService.removeNotificationWithNoteIdFromSystemBar(this, note.getId());
-        // mark the note as read if it's unread
-        if (note.isUnread()) {
-            NotificationsActions.markNoteAsRead(note);
-            note.setRead();
-            NotificationsTable.saveNote(note);
-            EventBus.getDefault().post(new NotificationEvents.NotificationsChanged());
-        }
+    private void markNoteAsRead(final Note note) {
+        new Thread(new Runnable() {
```

Comment:
I know we use this pattern a lot, however I believe it's not a best practice. The anonymous inner class keeps reference to its parent (`NotificationsDetailActivity`) and if the user leaves the activity (or rotates the screen) before the Thread finishes it's job, the activity leaks. We might consider creating a `static` inner class instead. Wdyt?